src/catalogue/app/src/main/java/catalogue/repositories/ProductRepository.java

```diff
@@ -3,13 +3,16 @@
 import catalogue.model.Product;
 import io.micronaut.core.annotation.NonNull;
 import io.micronaut.data.annotation.Join;
+import io.micronaut.data.jdbc.annotation.JdbcRepository;
 import io.micronaut.data.model.Pageable;
+import io.micronaut.data.model.query.builder.sql.Dialect;
 import io.micronaut.data.repository.PageableRepository;
 
 import javax.validation.constraints.NotNull;
 import java.util.List;
 import java.util.Optional;
 
+@JdbcRepository(dialect = Dialect.H2)
 public interface ProductRepository extends PageableRepository<Product, String> {
```

Comment:
What if we rename it to `H2ProductRepository ` or `InMemoryProductRepository `?

Reply:
The same comment as for CategoryRepository